### PR TITLE
Manually incrementing the hardcoded master version to 6.0

### PR DESCRIPTION
**Pull request by kobelb**
The `master` branch was hard-coded to be version 5.0 in the Version.js, this just increments that number.

The other approach that I was considering was adding in an optional 'branch to version' in the package.json to hopefully make it more obvious next time. Open to other alternative as well.

### CODE REVIEW

File: scripts/Version.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ Version.fromBranch = function (branch) {
   var m;
 
   // master === the highest version number
-  if (branch === 'master') return new Version('5.0.0');
+  if (branch === 'master') return new Version('6.0.0');
```
Comment (spalger) [inline scripts/Version.js]: Ugh... I'm not totally opposed to putting this in package.json's config section

**Comment by spalger** [comment]
I assume you also ran `npm run generate`?

**Comment by kobelb** [comment]
@spalger yup, I ran `npm run generate` and it only changed 5.x and master.

File: scripts/Version.js (hunk 1 above)
Comment (kobelb) [inline scripts/Version.js]: Cool, lemme throw something together for this!

**Comment by kobelb** [comment]
@spalger threw together a version that uses the package.json to drive this, lemme know what you think.

File: scripts/Version.js [hunk 2]
```diff
@@ -17,16 +18,17 @@ function Version(v) {
...
+  var majorMinorRegex = /^\d+\.\d+$/;
 
   // n.m -> n.m.0
-  if (m = branch.match(/^\d+\.\d+$/)) return new Version(branch + '.0');
+  if (m = branch.match(majorMinorRegex)) return new Version(branch + '.0');
 
   // n.x -> n.(maxVersion + 1).0
   if (m = branch.match(/^(\d+)\.x$/i)) return maxMinorVersion(m[1]).increment('minor');
 
+  var branchVersion = branchVersions[branch];
```
Comment (kobelb) [inline scripts/Version.js]: We might want to use this before the other cases.. on second thought, lemme change this...

**Comment by spalger** [comment]
Thanks @kobelb

**Comment by spalger** [comment]
BTW, I was debugging b345e0b13d0b00b5b37a861e4839eb31d6eb81eb and ended up defining the master branch translation just like we do the translation for `5.x`.

To get the next major version:
 - for the list of branches in `pkg.config.supported_es_branches` and `pkg.config.unstable_es_branches`
    - split branch name on `.`
    - take the first part
    - parse it to a number
 - filter out any `NaN`s (excludes "master" at this point)
 - sort versions descending
 - take the first version and add `1`

To get the next minor version within a major version:
 - for the list of branches in `pkg.config.supported_es_branches` and `pkg.config.unstable_es_branches`
    - split branch name on `.`
    - parse all parts to a number
    - take the first two parts
 - filter tuples:
    - where `tuple.length !== 2` (excludes "master")
    - where any values are `NaN` (excludes `.x` branches)
    - where `tuple[0]` is does not match the desired major version
 - sort tuples by `tuple[1]` descending
 - take `tuple[1]` from the first tuple and add `1`

**Pull request merged**